.github/workflows/base_alpine.yml

```diff
@@ -9,16 +9,48 @@ on:
       - 'docker/base/os/alpine/Dockerfile.alpine'
   workflow_dispatch:
 
+env:
+  PLATFORMS: linux/amd64,linux/arm64
+
 jobs:
-  deploy:
+  base:
     name: Checkout and deploy
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1  
-      - uses: docker/build-push-action@v1
+      - uses: actions/checkout@v1
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
         with:
             username: ${{ secrets.DOCKER_USERNAME }}
             password: ${{ secrets.DOCKER_TOKEN }}
-            repository: thevlang/vlang
-            tags: alpine-base
-            dockerfile: ${{github.workspace}}/docker/base/os/alpine/Dockerfile.alpine
\ No newline at end of file
+      - uses: docker/build-push-action@v4
+        with:
+            platforms: ${{ env.PLATFORMS}}
+            tags: ${{ secrets.DOCKER_USERNAME }}/vlang:base-alpine
+            push: true
+            file: docker/base/os/alpine/Dockerfile.alpine
+  vlang:
```

Comment:
We have this build in another docker file. They should be separated since base images have different release cycle than the build and runtime images. Could you separate the builds to allow matrix builds in the corresponding Dockerfile?

Reply:
is not possible to put some conditional to run one or the other for different life cicles? 

Also, when I tried to work with the docker images, one of the image was depending on the same image himself, maybe one day next week, you can explain it better why is done this way.